### PR TITLE
fix: broken patterns + examples

### LINKED ISSUE #761: triage failing examples
[first_ten_keys.ts]
```
vscode ➜ /workspaces/capi (example-tool-testing-2) $ deno task run examples/first_ten_keys.ts 
Task run deno task capi -- deno run -A -r=http://localhost:4646/ "examples/first_ten_keys.ts"
Task capi deno run -A main.ts --dbg "--" "deno" "run" "-A" "-r=http://localhost:4646/" "examples/first_ten_keys.ts"
Reusing existing Capi server at "http://localhost:4646/"
error: Uncaught (in promise) Unhandled {
  value: ScaleAssertError: !(value instanceof Uint8Array)
    at AssertState.instanceof (https://deno.land/x/scale@v0.11.0/common/assert.ts:37:13)
    at Codec._assert (https://deno.land/x/scale@v0.11.0/codecs/array.ts:104:24)
    at Codec._assert (file:///workspaces/capi/crypto/hashers.ts:52:14)
    at Codec._assert (file:///workspaces/capi/frame_metadata/key_codecs.ts:25:12)
    at Module.assert (https://deno.land/x/scale@v0.11.0/common/codec.ts:140:9)
    at RunMap.fn (file:///workspaces/capi/fluent/CodecRune.ts:8:9)
    at RunMap._evaluate (file:///workspaces/capi/rune/ValueRune.ts:153:40)
    at async RunMap.evaluate (file:///workspaces/capi/rune/Rune.ts:226:14)
    at async RunThrows._evaluate (file:///workspaces/capi/rune/ValueRune.ts:184:14)
    at async RunThrows.evaluate (file:///workspaces/capi/rune/Rune.ts:226:14)
}
```

[size_tree.ts]
```
vscode ➜ /workspaces/capi (example-tool-testing-2) $ deno task run examples/size_tree.ts
Task run deno task capi -- deno run -A -r=http://localhost:4646/ "examples/size_tree.ts"
Task capi deno run -A main.ts --dbg "--" "deno" "run" "-A" "-r=http://localhost:4646/" "examples/size_tree.ts"
Capi server listening at "http://localhost:4646/"
error: Uncaught (in promise) Unhandled {
  value: ScaleAssertError: !(value instanceof Uint8Array)
    at AssertState.instanceof (https://deno.land/x/scale@v0.11.0/common/assert.ts:37:13)
    at Codec._assert (https://deno.land/x/scale@v0.11.0/codecs/array.ts:104:24)
    at Codec._assert (file:///workspaces/capi/crypto/hashers.ts:52:14)
    at Codec._assert (file:///workspaces/capi/frame_metadata/key_codecs.ts:58:14)
    at Codec._assert (file:///workspaces/capi/frame_metadata/key_codecs.ts:25:12)
    at Module.assert (https://deno.land/x/scale@v0.11.0/common/codec.ts:140:9)
    at RunMap.fn (file:///workspaces/capi/fluent/CodecRune.ts:8:9)
    at RunMap._evaluate (file:///workspaces/capi/rune/ValueRune.ts:153:40)
    at async RunMap.evaluate (file:///workspaces/capi/rune/Rune.ts:226:14)
    at async RunThrows._evaluate (file:///workspaces/capi/rune/ValueRune.ts:184:14)
}
```

[virtual_multisig.ts]
```
vscode ➜ /workspaces/capi (example-tool-testing-2) $ deno task run examples/virtual_multisig.ts
Task run deno task capi -- deno run -A -r=http://localhost:4646/ "examples/virtual_multisig.ts"
Task capi deno run -A main.ts --dbg "--" "deno" "run" "-A" "-r=http://localhost:4646/" "examples/virtual_multisig.ts"
Capi server listening at "http://localhost:4646/"
Proxy creation: ready
Proxy creation: { inBlock: "0x026258f1bc883b0dce7f2c8d87dc70b4fd4c8ce0700eb08306226caa1c09ca6f" }
Proxy creation: { finalized: "0x026258f1bc883b0dce7f2c8d87dc70b4fd4c8ce0700eb08306226caa1c09ca6f" }
error: Uncaught (in promise) Unhandled {
  value: ScaleAssertError: !("type" in value.call.value.calls[0].value.dest)
    at AssertState.key (https://deno.land/x/scale@v0.11.0/common/assert.ts:45:13)
    at Codec._assert (https://deno.land/x/scale@v0.11.0/codecs/union.ts:57:32)
    at Codec._assert (https://deno.land/x/scale@v0.11.0/codecs/object.ts:16:14)
    at Codec._assert (https://deno.land/x/scale@v0.11.0/codecs/object.ts:79:16)
    at Codec._assert (https://deno.land/x/scale@v0.11.0/codecs/object.ts:79:16)
    at Codec._assert (https://deno.land/x/scale@v0.11.0/codecs/union.ts:62:76)
    at Codec._assert (https://deno.land/x/scale@v0.11.0/codecs/object.ts:16:14)
    at Codec._assert (https://deno.land/x/scale@v0.11.0/codecs/object.ts:79:16)
    at Codec._assert (https://deno.land/x/scale@v0.11.0/codecs/object.ts:79:16)
    at Codec._assert (https://deno.land/x/scale@v0.11.0/codecs/union.ts:62:76)
}
```

[polkadot_js_signer.ts]
```
vscode ➜ /workspaces/capi (example-tool-testing-2) $ deno task run examples/polkadot_js_signer.ts
Task run deno task capi -- deno run -A -r=http://localhost:4646/ "examples/polkadot_js_signer.ts"
Task capi deno run -A main.ts --dbg "--" "deno" "run" "-A" "-r=http://localhost:4646/" "examples/polkadot_js_signer.ts"
Capi server listening at "http://localhost:4646/"
error: Uncaught (in promise) Unhandled {
  value: ServerError: Invalid Transaction
    at RunMap.fn (file:///workspaces/capi/fluent/ConnectionRune.ts:53:32)
    at RunMap._evaluate (file:///workspaces/capi/rune/ValueRune.ts:153:40)
    at async RunMap.evaluate (file:///workspaces/capi/rune/Rune.ts:226:14)
    at async RunThrows._evaluate (file:///workspaces/capi/rune/ValueRune.ts:184:14)
    at async RunThrows.evaluate (file:///workspaces/capi/rune/Rune.ts:226:14)
    at async Promise.all (index 0)
    at async RunLs.evaluate (file:///workspaces/capi/rune/Rune.ts:226:14)
    at async RunMap._evaluate (file:///workspaces/capi/rune/ValueRune.ts:151:20)
    at async RunMap.evaluate (file:///workspaces/capi/rune/Rune.ts:226:14)
    at async RunFilter._evaluate (file:///workspaces/capi/rune/ValueRune.ts:264:21)
Caused by Invalid Transaction
}
```

[multisig_pure_proxy_stash.ts]
```
vscode ➜ /workspaces/capi (example-tool-testing-2) $ deno task run examples/multisig_pure_proxy_stash.ts
Task run deno task capi -- deno run -A -r=http://localhost:4646/ "examples/multisig_pure_proxy_stash.ts"
Task capi deno run -A main.ts --dbg "--" "deno" "run" "-A" "-r=http://localhost:4646/" "examples/multisig_pure_proxy_stash.ts"
Capi server listening at "http://localhost:4646/"
Fund Multisig: ready
Fund Multisig: { inBlock: "0xec120d19da761bcf121ac73ff998732488e3f8805fd24811ac87b8bbc529a870" }
Fund Multisig: { finalized: "0xec120d19da761bcf121ac73ff998732488e3f8805fd24811ac87b8bbc529a870" }
error: Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'when')
        value = value[key]
                     ^
    at RunMap.fn (file:///workspaces/capi/rune/ValueRune.ts:53:22)
    at RunMap._evaluate (file:///workspaces/capi/rune/ValueRune.ts:153:40)
    at async RunMap.evaluate (file:///workspaces/capi/rune/Rune.ts:226:14)
    at async RunCaptureUnhandled._evaluate (file:///workspaces/capi/rune/Rune.ts:370:14)
    at async RunCaptureUnhandled.evaluate (file:///workspaces/capi/rune/Rune.ts:226:14)
    at async Promise.all (index 6)
    at async RunLs.evaluate (file:///workspaces/capi/rune/Rune.ts:226:14)
    at async RunMap._evaluate (file:///workspaces/capi/rune/ValueRune.ts:151:20)
    at async RunMap.evaluate (file:///workspaces/capi/rune/Rune.ts:226:14)
    at async Promise.all (index 1)
```

[entry_page.ts]
```
vscode ➜ /workspaces/capi (example-tool-testing-2) $ deno task run examples/entry_page.ts               
Task run deno task capi -- deno run -A -r=http://localhost:4646/ "examples/entry_page.ts"
Task capi deno run -A main.ts --dbg "--" "deno" "run" "-A" "-r=http://localhost:4646/" "examples/entry_page.ts"
Capi server listening at "http://localhost:4646/"
error: Uncaught Unhandled {
  value: ScaleAssertError: typeof value !== "number"
    at AssertState.typeof (https://deno.land/x/scale@v0.11.0/common/assert.ts:25:13)
    at Codec._assert (https://deno.land/x/scale@v0.11.0/codecs/int.ts:37:20)
    at Codec._assert (file:///workspaces/capi/crypto/hashers.ts:52:14)
    at Codec._assert (file:///workspaces/capi/frame_metadata/key_codecs.ts:25:12)
    at Module.assert (https://deno.land/x/scale@v0.11.0/common/codec.ts:140:9)
    at RunMap.fn (file:///workspaces/capi/fluent/CodecRune.ts:8:9)
    at RunMap._evaluate (file:///workspaces/capi/rune/ValueRune.ts:153:40)
    at async RunMap.evaluate (file:///workspaces/capi/rune/Rune.ts:226:14)
    at async RunThrows._evaluate (file:///workspaces/capi/rune/ValueRune.ts:184:14)
    at async RunThrows.evaluate (file:///workspaces/capi/rune/Rune.ts:226:14)
}
```

### PR DESCRIPTION
**Pull request by harrysolovay**
Fixes #761

Fixes...

- broken patterns, including multisigs, virtual multisigs and size tree
- broken examples
- ws rpc close error

### CODE REVIEW

File: fluent/EventsRune.ts [hunk 1]
```diff
@@ -5,6 +5,21 @@ import { PatternRune } from "./PatternRune.ts"
 
 export class EventsRune<out C extends Chain, out U> extends PatternRune<Event<C>[], C, U> {}
 
+export interface TmpEventsChain extends Chain {
```
Comment (harrysolovay) [inline fluent/EventsRune.ts:8]: We'll remove this in a follow-up PR that tackles constraints

**Comment by ryanleecode** [comment]
`polkadot_js_signer is` still failing but wasn't modified in this PR. Not sure if its within scope.

```
error: Uncaught (in promise) Unhandled {
  value: ServerError: Invalid Transaction
    at RunMap.fn (file:///home/ryan/Documents/Repositories/capi/fluent/ConnectionRune.ts:53:32)
    at RunMap._evaluate (file:///home/ryan/Documents/Repositories/capi/rune/ValueRune.ts:150:40)
    at async RunMap.evaluate (file:///home/ryan/Documents/Repositories/capi/rune/Rune.ts:222:14)
    at async RunThrows._evaluate (file:///home/ryan/Documents/Repositories/capi/rune/ValueRune.ts:181:14)
    at async RunThrows.evaluate (file:///home/ryan/Documents/Repositories/capi/rune/Rune.ts:222:14)
    at async Promise.all (index 0)
    at async RunLs.evaluate (file:///home/ryan/Documents/Repositories/capi/rune/Rune.ts:222:14)
    at async RunMap._evaluate (file:///home/ryan/Documents/Repositories/capi/rune/ValueRune.ts:148:20)
    at async RunMap.evaluate (file:///home/ryan/Documents/Repositories/capi/rune/Rune.ts:222:14)
    at async RunFilter._evaluate (file:///home/ryan/Documents/Repositories/capi/rune/ValueRune.ts:261:21)
Caused by Invalid Transaction
}

finished polkadot_js_signer.ts
```

**Comment by ryanleecode** [comment]
sounds good

**Review by tjjfvi — DISMISSED** [review]

File: patterns/multisig/VirtualMultisigRune.ts [hunk 2]
```diff
@@ -44,41 +45,45 @@ export class VirtualMultisigRune<out C extends Chain, out U>
...
-  proxyBySenderAddr<X>(...[senderAddr]: RunicArgs<X, [Uint8Array]>) {
-    const signatories = this.inner.into(ValueRune).access("signatories")
-    return Rune
-      .tuple([signatories, this.proxies, senderAddr])
-      .map(([signatories, proxies, senderAddr]) =>
-        MultiAddress.Id(signatories[proxies[hex.encode(senderAddr)]!]!)
-      )
+  proxyBySenderAccountId<X>(...[senderAccountId]: RunicArgs<X, [senderAccountId: Uint8Array]>) {
+    const senderAccountIdHex = Rune.resolve(senderAccountId).map(hex.encode)
+    const senderProxyI = this.proxies.access(senderAccountIdHex)
```
Comment (ryanleecode) [inline patterns/multisig/VirtualMultisigRune.ts:50]: `senderProxyId` ?

File: patterns/sizeTree.ts [hunk 3]
```diff
@@ -11,10 +11,19 @@ export function sizeTree<C extends Chain, U, X>(
...
           Rune.rec(Object.fromEntries(
-            Object.values(pallet.storage).map((entry) => [
-              entry.name,
-              chain.pallet(pallet.name).storage(entry.name).size(undefined!, blockHash),
-            ]) || [],
+            Object.values(pallet.storage).map((entry) => {
+              const partialKeyType: string = entry.partialKey._metadata[0]?.args[2]?._metadata[0]
+                ?.name
+              const partialKey = {
+                $partialEmptyKey: undefined,
```
Comment (ryanleecode) [inline patterns/sizeTree.ts:18]: can omit since if you index it and the key doesnt exist it will return` undefined` anyways

File: rpc/ws.ts [hunk 4]
```diff
@@ -49,7 +49,28 @@ export class WsConnection extends Connection {
...
+        const controller = new AbortController()
+        const options: AddEventListenerOptions = {
+          signal: controller.signal,
+          once: true,
+        }
+        this.ws.addEventListener("open", () => {
+          this.ws.close()
+          controller.abort()
+        }, options)
+        this.ws.addEventListener("close", () => {}, options)
```
Comment (ryanleecode) [inline rpc/ws.ts:68]: do we need this if the handle block is ermpty?

File: patterns/sizeTree.ts (hunk 3 above)
Comment (tjjfvi) [reply patterns/sizeTree.ts:18]: `null` is a valid partial key for everything, so this lookup isn't needed at all.

File: rpc/ws.ts (hunk 4 above)
Comment (harrysolovay) [reply rpc/ws.ts:68]: No, this is a desirable outcome (as we're trying to close the socket anyways).

File: patterns/sizeTree.ts (hunk 3 above)
Comment (harrysolovay) [reply patterns/sizeTree.ts:18]: When changed to null we get the following:

Screen Shot 2023-03-24 at 12 15 58 PM

Comment (tjjfvi) [reply patterns/sizeTree.ts:18]: Ah, yes – `if (assert.value === null) return` should be added above https://github.com/paritytech/capi/blob/main/frame_metadata/key_codecs.ts#L84

File: patterns/multisig/VirtualMultisigRune.ts (hunk 2 above)
Comment (ryanleecode) [reply patterns/multisig/VirtualMultisigRune.ts:50]: this is still missing unless you have yet to push it

Comment (harrysolovay) [reply patterns/multisig/VirtualMultisigRune.ts:50]: If the GH review interface is acting wonky, perhaps view in the branch's tree?

Screen Shot 2023-03-24 at 1 06 22 PM

**Review by tjjfvi — APPROVED** [review]

**Review by ryanleecode — DISMISSED** [review]

**Review by tjjfvi — APPROVED** [review]

**Pull request merged**